src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -217,6 +285,251 @@ await contextualFilter.EvaluateAsync(context, appContext).ConfigureAwait(false)
             return enabled;
         }
 
```

Comment:
Updating the session state with feature enabled state needs to be the last operation done in the flow of a caller calling IsEnabledAsync